### ISSUE TITLE
serial.gesv_double failures with CLANG13+CUDA10 & C++17, with tpls enabled

### ISSUE THREAD

**Issue by e10harvey** [kokkos/kokkos-kernels#1525]
Example of failure output with clang13+cuda10 & c++17:

```
7: Test command: /path/to/unit_test/KokkosKernels_blas_serial
7: Test timeout computed to be: 1500
7: [==========] Running 177 tests from 1 test case.
7: [----------] Global test environment set-up.
7: [----------] 177 tests from serial
7: [ RUN      ] serial.gesv_double
1/1 Test #7: blas_serial ......................***Exception: SegFault  3.88 sec
[==========] Running 177 tests from 1 test case.
[----------] Global test environment set-up.
[----------] 177 tests from serial
[ RUN      ] serial.gesv_double


0% tests passed, 1 tests failed out of 1

Total Test time (real) =   3.90 sec

The following tests FAILED:
	  7 - blas_serial (SEGFAULT)
Errors while running CTest
```

## Reproducer
With [kokkos@4477a25ebe12b655cd5da273eec4ab954fbf32d5](https://github.com/kokkos/kokkos/tree/develop) and [kokkos-kernels@0f5c8cc57f366a902cf415b97898d2ed88de9d56](https://github.com/e10harvey/kokkos-kernels/tree/add_clang10_cuda10):
```
source /etc/profile.d/modules.sh
module purge
module load cmake/3.21.2 clang/13.0.0 openblas/0.3.16/gcc/7.2.0 cuda/10.1.243

$KOKKOSKERNELS_PATH/cm_generate_makefile.bash --with-devices=Cuda --arch=Power9,Volta70 --compiler=$GCC_PATH/llvm-13.0.0-t6hzufjroylzhs7hg3dvmhrrcsvhygzv/bin/clang++ --cxxflags="-O3 -Wall -Wunused-parameter -Wshadow -pedantic -Werror -Wsign-compare -Wtype-limits -Wuninitialized " --cxxstandard="17" --ldflags="" --with-cuda=$CUDA_PATH  --kokkos-path=$KOKKOS_PATH --kokkoskernels-path=$KOKKOSKERNELS_PATH --with-scalars='double,complex_double' --with-ordinals=int --with-offsets=int,size_t --with-layouts=LayoutLeft --with-tpls=blas --user-blas-path=$OPENBLAS_ROOT/lib --user-lapack-path=$OPENBLAS_ROOT/lib --user-blas-lib=blas --user-lapack-lib=lapack --extra-linker-flags=-lgfortran,-lm --with-options= --with-cuda-options=   --no-examples 

make -j32

ctest -V
```

**Comment by e10harvey** [comment]
@vqd8a: Would you have time to take a look?

**Comment by vqd8a** [comment]
@e10harvey I am looking at it.

**Comment by e10harvey** [comment]
Using the binaries I built when I discovered this issue I reproduced the bug via:
```
$ ./unit_test/KokkosKernels_blas_serial --gtest_filter=serial.gesv_double
Note: Google Test filter = serial.gesv_double
[==========] Running 1 test from 1 test case.
[----------] Global test environment set-up.
[----------] 1 test from serial
[ RUN      ] serial.gesv_double
Segmentation fault (core dumped
```

Upon listing the help menu I also noticed:
```
$ ./unit_test/KokkosKernels_blas_serial --help

--------------------------------------------------------------------------------
-------------Kokkos command line arguments--------------------------------------
--------------------------------------------------------------------------------
This program is using Kokkos.  You can use the following command line flags to
control its behavior:

Kokkos Core Options:
  --kokkos-help                  : print this message
  --kokkos-disable-warnings      : disable kokkos warning messages
  --kokkos-print-configuration   : print configuration
  --kokkos-tune-internals        : allow Kokkos to autotune policies and declare
                                   tuning features through the tuning system. If
                                   left off, Kokkos uses heuristics
  --kokkos-num-threads=INT       : specify total number of threads to use for
                                   parallel regions on the host.
  --kokkos-device-id=INT         : specify device id to be used by Kokkos.
  --kokkos-map-device-id-by=(random|mpi_rank)
                                 : strategy to select device-id automatically from
                                   available devices.
                                   - random:   choose a random device from available.
                                   - mpi_rank: choose device-id based on a round robin
                                               assignment of local MPI ranks.
                                               Works with OpenMPI, MVAPICH, SLURM, and
                                               derived implementations.

Kokkos Tools Options:
  --kokkos-tools-libs=STR        : Specify which of the tools to use. Must either
                                   be full path to library or name of library if the
                                   path is present in the runtime library search path
                                   (e.g. LD_LIBRARY_PATH)
  --kokkos-tools-help            : Query the (loaded) kokkos-tool for its command-line
                                   option support (which should then be passed via
                                   --kokkos-tools-args="...")
  --kokkos-tools-args=STR        : A single (quoted) string of options which will be
                                   whitespace delimited and passed to the loaded
                                   kokkos-tool as command-line arguments. E.g.
                                   `<EXE> --kokkos-tools-args="-c input.txt"` will
                                   pass `<EXE> -c input.txt` as argc/argv to tool

Except for --kokkos[-tools]-help, you can alternatively set the corresponding
environment variable of a flag (all letters in upper-case and underscores
instead of hyphens). For example, to disable warning messages, you can either
specify --kokkos-disable-warnings or set the KOKKOS_DISABLE_WARNINGS
environment variable to yes.

Join us on Slack, visit https://kokkosteam.slack.com
Report bugs to https://github.com/kokkos/kokkos/issues
--------------------------------------------------------------------------------

This program contains tests written using Google Test. You can use the
following command line flags to control its behavior:

Test Selection:
  --gtest_list_tests
      List the names of all tests instead of running them. The name of
      TEST(Foo, Bar) is "Foo.Bar".
  --gtest_filter=POSTIVE_PATTERNS[-NEGATIVE_PATTERNS]
      Run only the tests whose name matches one of the positive patterns but
      none of the negative patterns. '?' matches any single character; '*'
      matches any substring; ':' separates two patterns.
  --gtest_also_run_disabled_tests
      Run all disabled tests too.

Test Execution:
  --gtest_repeat=[COUNT]
      Run the tests repeatedly; use a negative count to repeat forever.
  --gtest_shuffle
      Randomize tests' orders on every iteration.
  --gtest_random_seed=[NUMBER]
      Random number seed to use for shuffling test orders (between 1 and
      99999, or 0 to use a seed based on the current time).

Test Output:
  --gtest_color=(yes|no|auto)
      Enable/disable colored output. The default is auto.
  --gtest_print_time=0
      Don't print the elapsed time of each test.
  --gtest_output=xml[:DIRECTORY_PATH/|:FILE_PATH]
      Generate an XML report in the given directory or with the given file
      name. FILE_PATH defaults to test_details.xml.
  --gtest_stream_result_to=HOST:PORT
      Stream test results to the given server.

Assertion Behavior:
  --gtest_death_test_style=(fast|threadsafe)
      Set the default death test style.
  --gtest_break_on_failure
      Turn assertion failures into debugger break-points.
  --gtest_throw_on_failure
      Turn assertion failures into C++ exceptions.
  --gtest_catch_exceptions=0
      Do not report exceptions as test failures. Instead, allow them
      to crash the program or throw a pop-up (on Windows).

Except for --gtest_list_tests, you can alternatively set the corresponding
environment variable of a flag (all letters in upper-case). For example, to
disable colored text output, you can either specify --gtest_color=no or set
the GTEST_COLOR environment variable to no.

For more information, please read the Google Test documentation at
http://code.google.com/p/googletest/. If you find a bug in Google Test
(not one in your own code or tests), please report it to
<googletestframework@googlegroups.com>.
OpenBLAS : munmap failed:: Invalid argument
error code=22,	release->address=0x2000dad19000
OpenBLAS : munmap failed:: Invalid argument
error code=22,	release->address=0x20007e4a2000
OpenBLAS : munmap failed:: Invalid argument
error code=22,	release->address=0x20005a012000
OpenBLAS : munmap failed:: Invalid argument
error code=22,	release->address=0x20013d777000
OpenBLAS : munmap failed:: Invalid argument
error code=22,	release->address=0x2000beb96000
OpenBLAS : munmap failed:: Invalid argument
error code=22,	release->address=0x2001fabff000
OpenBLAS : munmap failed:: Invalid argument
error code=22,	release->address=0x20019be58000
OpenBLAS : munmap failed:: Invalid argument
error code=22,	release->address=0x20003e709000
OpenBLAS : munmap failed:: Invalid argument
error code=22,	release->address=0x2001d799e000
OpenBLAS : munmap failed:: Invalid argument
error code=22,	release->address=0x200157bc3000
OpenBLAS : munmap failed:: Invalid argument
error code=22,	release->address=0x2002faa13000
OpenBLAS : munmap failed:: Invalid argument
error code=22,	release->address=0x20011d9ef000
OpenBLAS : munmap failed:: Invalid argument
error code=22,	release->address=0x2002bb975000
OpenBLAS : munmap failed:: Invalid argument
error code=22,	release->address=0x20009b1b7000
OpenBLAS : munmap failed:: Invalid argument
error code=22,	release->address=0x2001bb159000
OpenBLAS : munmap failed:: Invalid argument
error code=22,	release->address=0x200218de1000
OpenBLAS : munmap failed:: Invalid argument
error code=22,	release->address=0x2000161f8000
OpenBLAS : munmap failed:: Invalid argument
error code=22,	release->address=0x2002da3f7000
OpenBLAS : munmap failed:: Invalid argument
error code=22,	release->address=0x20031a886000
OpenBLAS : munmap failed:: Invalid argument
error code=22,	release->address=0x200238eb7000
OpenBLAS : munmap failed:: Invalid argument
error code=22,	release->address=0x2001782ea000
OpenBLAS : munmap failed:: Invalid argument
error code=22,	release->address=0x2000f64ab000
OpenBLAS : munmap failed:: Invalid argument
error code=22,	release->address=0x2003e2b8d000
OpenBLAS : munmap failed:: Invalid argument
error code=22,	release->address=0x200266ffe000
OpenBLAS : munmap failed:: Invalid argument
error code=22,	release->address=0x20042ae0c000
OpenBLAS : munmap failed:: Invalid argument
error code=22,	release->address=0x200b8db1a000
OpenBLAS : munmap failed:: Invalid argument
error code=22,	release->address=0x200fa2779000
OpenBLAS : munmap failed:: Invalid argument
error code=22,	release->address=0x2007a3f8c000
OpenBLAS : munmap failed:: Invalid argument
error code=22,	release->address=0x200fe3879000
OpenBLAS : munmap failed:: Invalid argument
error code=22,	release->address=0x200acaf48000
OpenBLAS : munmap failed:: Invalid argument
error code=22,	release->address=0x20043fbaa000
OpenBLAS : munmap failed:: Invalid argument
error code=22,	release->address=0x200c7a001000
OpenBLAS : munmap failed:: Invalid argument
error code=22,	release->address=0x200ffdebc000
OpenBLAS : munmap failed:: Invalid argument
error code=22,	release->address=0x2003bb9c5000
OpenBLAS : munmap failed:: Invalid argument
error code=22,	release->address=0x200278b75000
OpenBLAS : munmap failed:: Invalid argument
error code=22,	release->address=0x2005ee404000
OpenBLAS : munmap failed:: Invalid argument
error code=22,	release->address=0x2002a9001000
OpenBLAS : munmap failed:: Invalid argument
error code=22,	release->address=0x200814133000
OpenBLAS : munmap failed:: Invalid argument
error code=22,	release->address=0x200e32038000
OpenBLAS : munmap failed:: Invalid argument
error code=22,	release->address=0x2004def65000
OpenBLAS : munmap failed:: Invalid argument
error code=22,	release->address=0x20057f742000
OpenBLAS : munmap failed:: Invalid argument
error code=22,	release->address=0x20076355c000
OpenBLAS : munmap failed:: Invalid argument
error code=22,	release->address=0x200d4eaff000
OpenBLAS : munmap failed:: Invalid argument
error code=22,	release->address=0x200bfafd6000
OpenBLAS : munmap failed:: Invalid argument
error code=22,	release->address=0x200884d23000
OpenBLAS : munmap failed:: Invalid argument
error code=22,	release->address=0x200d9d6f7000
OpenBLAS : munmap failed:: Invalid argument
error code=22,	release->address=0x20033cb05000
OpenBLAS : munmap failed:: Invalid argument
error code=22,	release->address=0x200b6b4d5000
OpenBLAS : munmap failed:: Invalid argument
error code=22,	release->address=0x20037c44d000
OpenBLAS : munmap failed:: Invalid argument
error code=22,	release->address=0x200f8083e000
OpenBLAS : munmap failed:: Invalid argument
error code=22,	release->address=0x200cb9e38000
OpenBLAS : munmap failed:: Invalid argument
error code=22,	release->address=0x200c0c5f5000
OpenBLAS : munmap failed:: Invalid argument
error code=22,	release->address=0x200702111000
OpenBLAS : munmap failed:: Invalid argument
error code=22,	release->address=0x200aeb23d000
OpenBLAS : munmap failed:: Invalid argument
error code=22,	release->address=0x200b2b224000
OpenBLAS : munmap failed:: Invalid argument
error code=22,	release->address=0x200f01bd9000
OpenBLAS : munmap failed:: Invalid argument
error code=22,	release->address=0x200f5da95000
OpenBLAS : munmap failed:: Invalid argument
error code=22,	release->address=0x2009b7533000
OpenBLAS : munmap failed:: Invalid argument
error code=22,	release->address=0x200babfad000
OpenBLAS : munmap failed:: Invalid argument
error code=22,	release->address=0x200ee17dc000
OpenBLAS : munmap failed:: Invalid argument
error code=22,	release->address=0x2006008b9000
OpenBLAS : munmap failed:: Invalid argument
error code=22,	release->address=0x200e97175000
OpenBLAS : munmap failed:: Invalid argument
error code=22,	release->address=0x20054e076000
OpenBLAS : munmap failed:: Invalid argument
error code=22,	release->address=0x200640697000
OpenBLAS : munmap failed:: Invalid argument
error code=22,	release->address=0x200e1feae000
OpenBLAS : munmap failed:: Invalid argument
error code=22,	release->address=0x20051fb19000
OpenBLAS : munmap failed:: Invalid argument
error code=22,	release->address=0x200946d13000
OpenBLAS : munmap failed:: Invalid argument
error code=22,	release->address=0x200d1df07000
OpenBLAS : munmap failed:: Invalid argument
error code=22,	release->address=0x20055ead9000
OpenBLAS : munmap failed:: Invalid argument
error code=22,	release->address=0x2004c55dd000
OpenBLAS : munmap failed:: Invalid argument
error code=22,	release->address=0x20035b34c000
OpenBLAS : munmap failed:: Invalid argument
error code=22,	release->address=0x200f39ae6000
OpenBLAS : munmap failed:: Invalid argument
error code=22,	release->address=0x200fc0fe7000
OpenBLAS : munmap failed:: Invalid argument
error code=22,	release->address=0x20074da51000
OpenBLAS : munmap failed:: Invalid argument
error code=22,	release->address=0x200a9671a000
OpenBLAS : munmap failed:: Invalid argument
error code=22,	release->address=0x200e759b8000
OpenBLAS : munmap failed:: Invalid argument
error code=22,	release->address=0x20050dbd6000
OpenBLAS : munmap failed:: Invalid argument
error code=22,	release->address=0x200ab75df000
OpenBLAS : munmap failed:: Invalid argument
error code=22,	release->address=0x200b5ac6b000
OpenBLAS : munmap failed:: Invalid argument
error code=22,	release->address=0x200844cc5000
OpenBLAS : munmap failed:: Invalid argument
error code=22,	release->address=0x20046cbec000
OpenBLAS : munmap failed:: Invalid argument
error code=22,	release->address=0x2006b0fa1000
OpenBLAS : munmap failed:: Invalid argument
error code=22,	release->address=0x200873658000
OpenBLAS : munmap failed:: Invalid argument
error code=22,	release->address=0x20066170e000
OpenBLAS : munmap failed:: Invalid argument
error code=22,	release->address=0x200ddf992000
OpenBLAS : munmap failed:: Invalid argument
error code=22,	release->address=0x200c2c9ed000
OpenBLAS : munmap failed:: Invalid argument
error code=22,	release->address=0x2006d0288000
OpenBLAS : munmap failed:: Invalid argument
error code=22,	release->address=0x200cdbf24000
OpenBLAS : munmap failed:: Invalid argument
error code=22,	release->address=0x2009f7ffe000
OpenBLAS : munmap failed:: Invalid argument
error code=22,	release->address=0x2007229c1000
OpenBLAS : munmap failed:: Invalid argument
error code=22,	release->address=0x2008d5056000
OpenBLAS : munmap failed:: Invalid argument
error code=22,	release->address=0x2004ad225000
OpenBLAS : munmap failed:: Invalid argument
error code=22,	release->address=0x20078a858000
OpenBLAS : munmap failed:: Invalid argument
error code=22,	release->address=0x20082ab2c000
OpenBLAS : munmap failed:: Invalid argument
error code=22,	release->address=0x2006e1e5d000
OpenBLAS : munmap failed:: Invalid argument
error code=22,	release->address=0x20062b05c000
OpenBLAS : munmap failed:: Invalid argument
error code=22,	release->address=0x200987806000
OpenBLAS : munmap failed:: Invalid argument
error code=22,	release->address=0x2008f5f4d000
OpenBLAS : munmap failed:: Invalid argument
error code=22,	release->address=0x2007f355c000
OpenBLAS : munmap failed:: Invalid argument
error code=22,	release->address=0x200bda5c8000
OpenBLAS : munmap failed:: Invalid argument
error code=22,	release->address=0x200dfcddd000
OpenBLAS : munmap failed:: Invalid argument
error code=22,	release->address=0x200681abe000
OpenBLAS : munmap failed:: Invalid argument
error code=22,	release->address=0x2007d3642000
OpenBLAS : munmap failed:: Invalid argument
error code=22,	release->address=0x2009d7b4b000
OpenBLAS : munmap failed:: Invalid argument
error code=22,	release->address=0x200d3e13d000
OpenBLAS : munmap failed:: Invalid argument
error code=22,	release->address=0x2008b4e3f000
OpenBLAS : munmap failed:: Invalid argument
error code=22,	release->address=0x2005bf9dd000
OpenBLAS : munmap failed:: Invalid argument
error code=22,	release->address=0x200a68cb1000
OpenBLAS : munmap failed:: Invalid argument
error code=22,	release->address=0x200e50ac4000
OpenBLAS : munmap failed:: Invalid argument
error code=22,	release->address=0x200a1839c000
OpenBLAS : munmap failed:: Invalid argument
error code=22,	release->address=0x20093615e000
OpenBLAS : munmap failed:: Invalid argument
error code=22,	release->address=0x200c9cac7000
OpenBLAS : munmap failed:: Invalid argument
error code=22,	release->address=0x200967043000
OpenBLAS : munmap failed:: Invalid argument
error code=22,	release->address=0x20047d28b000
OpenBLAS : munmap failed:: Invalid argument
error code=22,	release->address=0x2005aeea7000
OpenBLAS : munmap failed:: Invalid argument
error code=22,	release->address=0x200f2223c000
OpenBLAS : munmap failed:: Invalid argument
error code=22,	release->address=0x200cfd82b000
OpenBLAS : munmap failed:: Invalid argument
error code=22,	release->address=0x200ec1623000
OpenBLAS : munmap failed:: Invalid argument
error code=22,	release->address=0x200d7e877000
```

@ndellingwood, @vqd8a, @kyungjoo-kim: Have you seen this before?

Does this indicate a problem in unmapping virtual addresses with the underlying OpenBlas TPL I've added in #1524?

Note that I am using a openblas library built with gcc/7.2.0 but clang/13.0.0 loads gcc/7.4.0 as a dependency:
```
$ module list
Currently Loaded Modulefiles:
 1) cmake/3.21.2   2) binutils/2.30.0   3) gcc/7.4.0   4) clang/13.0.0   5) openblas/0.3.16/gcc/7.2.0   6) cuda/10.1.243
 ```

**Comment by e10harvey** [comment]
Swapping out `openblas/0.3.16/gcc/7.2.0` with `openblas/0.3.20/gcc/9.3.0` shows:
```
$ ../scripts/cm_test_all_sandia --spot-check-tpls clang/13.0.0 --kokkos-path=../../kokkos --kokkoskernels-path=../ --cxxstandard=17
Running on machine: weaver

KokkosKernels Repository Status:  ead1baafa75a9c67b4853d9101575b0d953e09ce scripts/cm_test_all_sandia: Add initial clang+cuda build

Kokkos Repository Status:  4477a25ebe12b655cd5da273eec4ab954fbf32d5 Merge pull request #5445 from dalg24/openacc_shared_allocation_record_header


Going to test compilers:  clang/13.0.0
Testing compiler clang/13.0.0
Unrecognized compiler clang/13.0.0 when looking for Spack variants
Unrecognized compiler clang/13.0.0 when looking for Spack variants
Unrecognized compiler clang/13.0.0 when looking for Spack variants
  Starting job clang-13.0.0-Cuda-release
kokkos devices: Cuda
kokkos arch: Power9,Volta70
kokkos options: 
kokkos cuda options: 
kokkos cxxflags: -O3 -Wall -Wunused-parameter -Wshadow -pedantic -Werror -Wsign-compare -Wtype-limits -Wuninitialized 
extra_args: 
kokkoskernels scalars: 'double,complex_double'
kokkoskernels ordinals: int
kokkoskernels offsets: int,size_t
kokkoskernels layouts: LayoutLeft
  PASSED clang-13.0.0-Cuda-release
#######################################################
PASSED TESTS
#######################################################
clang-13.0.0-Cuda-release build_time=895 run_time=782
```

**Comment by e10harvey** [comment]
Fixed via [be1f6c9](https://github.com/e10harvey/kokkos-kernels/commit/be1f6c9e03ac21644e0c93e983ba3552653a3481).

**Comment by vqd8a** [comment]
@e10harvey Thanks for looking at this issue.
